src/Compilers/Core/Portable/Symbols/Attributes/AttributeDescription.cs

```diff
@@ -28,6 +28,12 @@ public AttributeDescription(string @namespace, string name, byte[][] signatures,
             this.MatchIgnoringCase = matchIgnoringCase;
         }
 
+        public bool Equals(AttributeDescription other)
```

Comment:
>public bool Equals(AttributeDescription other) [](start = 8, length = 46)

Why is this method needed? Implementing equality that works well for everybody is not a trivial thing. If you need some kind of comparison for the purposes of specific component, it is much better to define a custom comparer locally.